core/src/main/java/com/sequenceiq/cloudbreak/service/stack/connector/openstack/OpenStackMetadataSetup.java

```diff
@@ -139,26 +141,42 @@ private CoreInstanceMetaData createCoreMetaData(Stack stack, Server server, Stri
 
     @Override
     public InstanceSyncState getState(Stack stack, String instanceId) {
-        OSClient osClient = openStackUtil.createOSClient(stack);
-        Resource heatResource = stack.getResourceByType(ResourceType.HEAT_STACK);
-        String heatStackId = heatResource.getResourceName();
-        InstanceSyncState instanceSyncState = IN_PROGRESS;
-        org.openstack4j.model.heat.Stack heatStack = osClient.heat().stacks().getDetails(stack.getName(), heatStackId);
-        List<Map<String, Object>> outputs = heatStack.getOutputs();
+        try {
+            OSClient osClient = openStackUtil.createOSClient(stack);
+            Resource heatResource = stack.getResourceByType(ResourceType.HEAT_STACK);
+            String heatStackId = heatResource.getResourceName();
+            org.openstack4j.model.heat.Stack heatStack = osClient.heat().stacks().getDetails(stack.getName(), heatStackId);
+            List<Map<String, Object>> outputs = heatStack.getOutputs();
+            return getInstanceSyncState(instanceId, osClient, outputs);
+        } catch (Exception ex) {
+            throw new OpenStackResourceException("Failed to retrieve state of instance " + instanceId, ex);
```

Comment:
is it possible that (like on AWS) after the instance is deleted it won't be reported by OpenStack as DELETED but some kind of NotFound exception is thrown? If yes the exception should be examined here as well and return DELETED if it was a not found.